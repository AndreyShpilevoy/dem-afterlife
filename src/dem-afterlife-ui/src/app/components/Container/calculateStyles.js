import {createMediaQueryMin} from 'utils';

const calculateStyles = ({themeName, grid}) =>
    grid.containers.reduce((previous, {width, mediaMinString}) => ({
        ...previous,
        ...createMediaQueryMin(mediaMinString, {
            container: {
                width,
                marginLeft: 'auto',
                marginRight: 'auto'
            }
        })
    }), {options: {meta: 'Container', themeName} });

export default calculateStyles;