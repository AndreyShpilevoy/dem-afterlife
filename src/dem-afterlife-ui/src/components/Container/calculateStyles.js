import { createMediaQueryMin } from "utils";

const calculateStyles = ({ grid }) =>
  grid.containers.reduce(
    (previous, { width, mediaMinString }) => ({
      ...previous,
      ...createMediaQueryMin(mediaMinString, {
        container: {
          width,
          marginLeft: "auto",
          marginRight: "auto"
        }
      })
    }),
    {}
  );

export default calculateStyles;
