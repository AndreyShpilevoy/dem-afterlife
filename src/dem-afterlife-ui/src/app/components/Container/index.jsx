import React from 'react';
import {node, string, func, shape} from 'prop-types';
import {withStyles} from 'styles';
import {defaults} from 'utils';
import calculateStyles from './calculateStyles';

export const ContainerPure = ({children, styles, className, onClick}) => {
    const localeClassName = `${styles.container} ${className || defaults.emptyString}`;
    return (
        <div className={localeClassName} onClick={onClick}>
            {children}
        </div>
    );
};

ContainerPure.propTypes = {
    className: string,
    styles: shape().isRequired,
    children: node.isRequired,
    onClick: func
};

ContainerPure.defaultProps = {
    className: defaults.emptyString,
    onClick: defaults.emptyFunction
};

export default withStyles(theme => calculateStyles(theme))(ContainerPure);