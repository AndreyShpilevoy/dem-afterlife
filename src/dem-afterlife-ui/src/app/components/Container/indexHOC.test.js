/* eslint no-undef: 0, fp/no-unused-expression: 0, fp/no-nil: 0, fp/no-mutation:0*/

import React from 'react';
import {shallow} from 'enzyme';
import * as ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
import Container from './index';

ThemedStyleSheet.default.get = () => {};

describe('Container HOC', () => {
    const hocProps = {
        theme: 'default'
    };

    it('component match expected snapshot', () => {
        expect(shallow(<Container theme={hocProps.theme} />)).toMatchSnapshot();
    });
});
