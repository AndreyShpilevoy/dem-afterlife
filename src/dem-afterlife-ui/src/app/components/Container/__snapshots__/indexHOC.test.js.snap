// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container HOC component match expected snapshot 1`] = `
<ContainerPure
  className=""
  styles={
    Object {
      "container": "default-Container-container",
    }
  }
/>
`;
